### PR TITLE
Viro-terror grenade renovation

### PR DESCRIPTION
**Pull request by LanCartwright**
I swear if this fails I'm gonna start drinking

### CODE REVIEW

**Comment by Iamgoofball** [comment]
>better be gpd da,m gicco at thos [pomt

i think you already did

File: code/datums/diseases/tuberculosis.dm [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+					affected_mob.sleeping += 5
+			if(prob(2))
+				affected_mob << "<span class='userdanger'>You feel your mind relax and your thoughts drift!</span>"
+				affected_mob.confused = min(100, M.confused + 8)
+			if(prob(15)
+				affected_mob.vomit(0, 1)
+			if(prob(3))
+				affected_mob << "<span class='warning'><i>[pick("Your stomach silently rumbles...", "Your stomach seizes up and falls limp, muscles dead and lifeless.", "You could eat a crayon")]</i></span>"
+				affected_mob.overeatduration = max(affected_mob.overeatduration - 100, 0)
+				affected_mob.nutrition = max(M.nutrition - 100, 0
```
Comment (KazeEspada) [inline code/datums/diseases/tuberculosis.dm]: Missing )

**Comment by RemieRichards** [comment]
>start drinking

you already drink!

File: code/datums/diseases/tuberculosis.dm [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			if(prob(2))
+				affected_mob << "<span class='userdanger'>[pick("You feel your heart slowing...", "You slow your heartbeat.")]</span>"
+				affected_mob.adjustStaminaLoss(40)
+				if(affected_mob.getStaminaLoss() > 60 && !M.stat)
+					affected_mob.visible_message("<span class='warning'>[M] faints!</span>", "<span class='userdanger'>You surrender yourself and feel at peace...</span>")
+					affected_mob.sleeping += 5
+			if(prob(2))
+				affected_mob << "<span class='userdanger'>You feel your mind relax and your thoughts drift!</span>"
+				affected_mob.confused = min(100, M.confused + 8)
+			if(prob(15)
```
Comment (Fox-McCloud) [inline code/datums/diseases/tuberculosis.dm]: if(prob(15))

**Comment by Zilenan91** [comment]
So exactly how lethal is this fungal tuberculosis.

**Comment by Jordie0608** [comment]
Needs a changelog

**Comment by LanCartwright** [comment]
wilco when I'm sober

File: code/datums/uplink_item.dm [hunk 3]
```diff
@@ -549,6 +557,13 @@ var/list/uplink_items = list()
 	cost = 1
 	surplus = 50
 
+/datum/uplink_item/stealthy_weapons/traitor_virus_kit
+	name = "Virus Kit"
+	desc = "An active fungal pathogen in a sterile, compact box. Comes with one Bio Virus Antidote Kit (BVAK) autoinjector for rapid application on upto two targets each, a syringe, and a bottle containing the BVAK solution."
+	item = /obj/item/weapon/storage/box/syndie_kit/tuberculosiskit
+	cost = 9
+	surplus = 50
```
Comment (tkdrg) [inline code/datums/uplink_item.dm]: Needs a gamemode restriction here too

File: code/datums/uplink_item.dm [hunk 4]
```diff
@@ -292,6 +292,14 @@ var/list/uplink_items = list()
 	gamemodes = list(/datum/game_mode/nuclear,/datum/game_mode/gang)
 	surplus = 0
 
+/datum/uplink_item/stealthy_weapons/operator_virus_grenade
+	name = "Fungal Tuberculosis grenade"
+	desc = "A primed bio-grenade packed into a compact box. Comes with five Bio Virus Antidote Kit (BVAK) autoinjectors for rapid application on upto two targets each, a syringe, and a bottle containing the BVAK solution."
```
Comment (tkdrg) [inline code/datums/uplink_item.dm]: up to, not upto

File: code/datums/uplink_item.dm [hunk 5]
```diff
@@ -549,6 +557,13 @@ var/list/uplink_items = list()
 	cost = 1
 	surplus = 50
 
+/datum/uplink_item/stealthy_weapons/traitor_virus_kit
+	name = "Virus Kit"
+	desc = "An active fungal pathogen in a sterile, compact box. Comes with one Bio Virus Antidote Kit (BVAK) autoinjector for rapid application on upto two targets each, a syringe, and a bottle containing the BVAK solution."
```
Comment (tkdrg) [inline code/datums/uplink_item.dm]: up to, not upto

File: code/game/objects/items/weapons/storage/uplink_kits.dm [hunk 6]
```diff
@@ -240,3 +240,32 @@
...
+/obj/item/weapon/storage/box/virus_kit/tuberculosisgrenade/New()
+ 	..()
+ 	new /obj/item/weapon/grenade/chem_grenade/tuberculosis
+ 	new /obj/item/weapon/reagent_containers/hypospray/medipen/tuberculosiscure
+ 	new /obj/item/weapon/reagent_containers/hypospray/medipen/tuberculosiscure
+ 	new /obj/item/weapon/reagent_containers/hypospray/medipen/tuberculosiscure
+ 	new /obj/item/weapon/reagent_containers/hypospray/medipen/tuberculosiscure
+ 	new /obj/item/weapon/reagent_containers/hypospray/medipen/tuberculosiscure
+ 	new /obj/item/weapon/reagent_containers/syringe
+ 	new	/obj/item/weapon/reagent_containers/glass/bottle/tuberculosiscure
```
Comment (tkdrg) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: why the random tab

File: code/game/objects/items/weapons/storage/uplink_kits.dm [hunk 7]
```diff
@@ -240,3 +240,32 @@
...
+
+/obj/item/weapon/storage/box/syndie_kit/tuberculosiskit
+ 	name = "boxed virus kit"
+
+/obj/item/weapon/storage/box/virus_kit/tuberculosiskit/New()
+ 	..()
+ 	new /obj/item/weapon/reagent_containers/glass/bottle/tuberculosis
+ 	new /obj/item/weapon/reagent_containers/hypospray/medipen/tuberculosiscure
+ 	new /obj/item/weapon/reagent_containers/syringe
+ 	new	/obj/item/weapon/reagent_containers/glass/bottle/tuberculosiscure
```
Comment (tkdrg) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: why the random tab x2

File: code/game/objects/items/weapons/storage/uplink_kits.dm (hunk 6 above)
Comment (LanCartwright) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: I don't, actually know.

Comment (LanCartwright) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: actually I do. I hit tab instead of space

File: code/datums/uplink_item.dm (hunk 3 above)
Comment (LanCartwright) [inline code/datums/uplink_item.dm]: That's the virus kit, I think the virus kit shouldn't be restricted to operator only.

**Comment by Jordie0608** [comment]
>Needs a changelog

plz

**Pull request merged**